src/_utils/range.js

```diff
@@ -0,0 +1,20 @@
+/* Generates a semi-open range. */
+export default function* range(start, end, step = 1) {
+  if (end == null) {
+    end = start;
+    start = 0;
+  }
+
+  if (step === 0) {
+    console.error('Range must have a non-zero step.');
```

Comment:
Perhaps a `throw` would be more appropriate in a function?

Reply:
Not here :) idk, this is something so utilitary, I really don't see any need for exceptions

Reply:
Who said utility functions can't throw exceptions? If you can be absolutely sure it will never be called with `step = 0`, then the check is useless. Otherwise, exceptions seem like the appropriate error-reporting technique, and not the console